system/lib/fetch/emscripten_fetch.cpp

```diff
@@ -148,7 +148,13 @@ emscripten_fetch_t* emscripten_fetch(emscripten_fetch_attr_t* fetch_attr, const
 
 #undef STRDUP_OR_ABORT
 
-#if __EMSCRIPTEN_PTHREADS__
+// In asm.js we can use a fetch worker, which is created from the main asm.js
+// code. That lets us do sync operations by blocking on the worker etc.
+// In the wasm backend we don't have a fetch worker implemented yet, however,
+// we can still do basic synchronous fetches in the same places: if we can
+// block on another thread then we aren't the main thread, and if we aren't
+// the main thread then synchronous xhrs are legitimate.
+#if __EMSCRIPTEN_PTHREADS__ && !defined(__wasm__)
```

Comment:
@VirtualTim here is where the code paths diverge - note how in asm.js mode, with pthreads, we always use the worker to do things for us, by calling `proxy_fetch`.

This PR makes the wasm backend path avoid that, and instead do the fetch directly, using `start_fetch` below.

Reply:
Yeah that makes sense.